docs/REST API/rest-api.md

```diff
@@ -2504,7 +2504,7 @@ the `software` table.
 | bootstrap_package       | string | query | _Available in Fleet Premium_. Filters the hosts by the status of the MDM bootstrap package on the host. Valid options are 'installed', 'pending', or 'failed'. |
 | os_settings          | string  | query | Filters the hosts by the status of the operating system settings applied to the hosts. Valid options are 'verified', 'verifying', 'pending', or 'failed'. **Note: If this filter is used in Fleet Premium without a team ID filter, the results include only hosts that are not assigned to any team.** |
 | os_settings_disk_encryption | string | query | Filters the hosts by the status of the disk encryption setting applied to the hosts. Valid options are 'verified', 'verifying', 'action_required', 'enforcing', 'failed', or 'removing_enforcement'.  **Note: If this filter is used in Fleet Premium without a team ID filter, the results include only hosts that are not assigned to any team.** |
-| populate_software     | boolean | query | If `true`, the response will include a list of installed software for each host, including vulnerability data. |
+| populate_software     | boolean | query | If `true`, the response will include a list of installed software for each host, including vulnerability data. **This fetches a significant amount of data per host, causing significant CPU and RAM usage, so pay attention to page size and request concurrency when turning this on.** |
```

Comment:
I feel like we could go a little subtler with this message. Thinking if we go parenthesis instead of bold, it still draws attention to it but in a less "WARNING!" way; also thinking we can soften the language a bit. WDYT? (There's a fine line between "be mindful of how you're using this feature" and "be careful!! you probably shouldn't use this feature!")

```suggestion
| populate_software     | boolean | query | If `true`, the response will include a list of installed software for each host, including vulnerability data. (Note that software lists can be large, so this may cause significant CPU and RAM usage depending on page size and request concurrency.) |
```

Reply:
Fair enough. See context in the linked issue; in that case the customer ran at least one (probably multiple) web tier containers OOM by hitting this endpoint at a decent level of concurrency. The next time that happens we'll push the priority on that issue up, assuming some other source of OOM errors doesn't happen first.